apps/communications/dialer/js/call_log_db.js

```diff
@@ -672,8 +672,12 @@ var CallLogDBManager = {
       groupsStore.get(groupId).onsuccess = function onsuccess() {
         var group = this.result;
         if (group) {
-          self._updateExistingGroup(group, recentCall, groupsStore, callback);
+          self._updateExistingGroup(group, recentCall, txn, callback);
         } else {
+          // When creating a new group we don't reuse the outer transaction for
+          // accessing the database as we need to do further asynchronous
+          // operations that would make it commit before we get a chance to
+          // use it again.
           self._createNewGroup(groupId, recentCall, callback);
```

Comment:
``_createNewGroup()`` invokes the ``Contacts.findByNumber()`` function which is asynchronous. Since there's no more requests associated with the outer transaction in the event-loop, it completes automatically. Note that I cannot chain the new operations on the outer transaction precisely because of this, when ``Contacts.findByNumber()`` invokes its asynchronous callback the outer transaction has already been completed.